packages/wrangler/src/dev.tsx

```diff
@@ -912,7 +916,10 @@ export async function validateDevServerSettings(
 	// we want it to exit the Wrangler process early to allow the user to fix it. Calling it here forces
 	// the error to be thrown where it will correctly exit the Wrangler process
 	if (args.remote) {
-		await getZoneIdForPreview(host, routes);
+		const accountId =
+			args.accountId ?? config.account_id ?? (await getAccountId());
```

Comment:
@RamIdeas does this look right to you?

Reply:
Looks about right. I just checked on the `--x-dev-env` code path and it passes accountId through correctly so this will continue to work when we flip that flag on by default